src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/ReadOnlyTagHelperAttributeList.cs

```diff
@@ -8,67 +8,66 @@
 namespace Microsoft.AspNetCore.Razor.TagHelpers
 {
     /// <summary>
-    /// A read-only collection of <typeparamref name="TAttribute"/>s.
+    /// A read-only collection of <typeparamref name="TagHelperAttribute"/>s.
     /// </summary>
-    /// <typeparam name="TAttribute">
-    /// The type of <see cref="IReadOnlyTagHelperAttribute"/>s in the collection.
+    /// <typeparam name="TagHelperAttribute">
+    /// The type of <see cref="TagHelperAttribute"/>s in the collection.
     /// </typeparam>
-    public class ReadOnlyTagHelperAttributeList<TAttribute> : IReadOnlyList<TAttribute>
-        where TAttribute : IReadOnlyTagHelperAttribute
+    public class ReadOnlyTagHelperAttributeList : IReadOnlyList<TagHelperAttribute>
```

Comment:
I may have mentioned this before, but consider inheriting `ReadOnlyCollection<T>` to get rid of some of this boilerplate